### PR TITLE
Modify `findNotes()` to use V16 `searchNotes()` function

### PR DESCRIPTION
**Pull request by viciousAegis**
Modify SearchNotes to be able to use both legacy and V16 functions

## Purpose / Description
Add support for using new backend method `searchNotes()`

## Approach
By checking the schema version, the newer function is used on using V16. The newer `searchNotes()` function includes a parameter for sorting, which is not supported in legacy.

`findNotes()` function is modified in `Collection` so that V16 can be used when enabled

## How Has This Been Tested?

On my phone by enabling V16 backend in settings.

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by Arthur-Milchior** [comment]
As a side note, I'd appreciate if you could remove part of the template that you don't fill, just so that I know what not to read and what information is relevant.

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.kt [hunk 1]
```diff
@@ -1351,6 +1351,10 @@ open class Collection(
 
     /** Return a list of note ids  */
     fun findNotes(query: String?): List<Long> {
+        return findNotes(query, SortOrder.NoOrdering())
+    }
+
+    open fun findNotes(query: String?, order: SortOrder): List<Long> {
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Collection.kt:1357]:
```
    @JvmOverloads
    open fun findNotes(query: String?, order: SortOrder = SortOrder.NoOrdering()): List<Long> {
```
Kotlin has default parameter. I believe in this case it's more relevant than overriding

**Review by Arthur-Milchior — APPROVED** [review]

File: AnkiDroid/src/main/java/com/ichi2/libanki/CollectionV16.kt [hunk 2]
```diff
@@ -156,6 +156,27 @@ class CollectionV16(
...
     }
 
+    override fun findNotes(
+        query: String?,
+        order: SortOrder,
+    ): List<Long> {
+        val adjustedOrder = if (order is SortOrder.UseCollectionOrdering) {
+            @Suppress("DEPRECATION")
+            SortOrder.BuiltinSortKind(
+                get_config("sortType", null as String?) ?: "noteFld",
```
Comment (dae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/CollectionV16.kt:166]:
```suggestion
                get_config("noteSortType", null as String?) ?: "noteFld",
```

Notes mode uses different ordering key - found in collection.py in desktop code.

Comment (dae) [reply AnkiDroid/src/main/java/com/ichi2/libanki/CollectionV16.kt:166]: Also, does AnkiDroid expose a way to change the value of that config entry?

Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/CollectionV16.kt:166]: For cards yes. For note no. But I assume the same code can be used depending on card or note mode.

**Review by mikehardy — APPROVED** [review]
Looks good to me, I'd like a look from @dae now
Also, not sure if this idea excites you or not @viciousAegis but I'd love for the V16 developer setting to be turned in to a toggle + preference for persistence as a separate thing. I believe we are at the point where easier testing would be a good idea, then hopefully near future we can move it from developer area to advanced

File: AnkiDroid/src/main/java/com/ichi2/libanki/CollectionV16.kt [hunk 3]
```diff
@@ -156,6 +156,27 @@ class CollectionV16(
...
 
+    override fun findNotes(
+        query: String?,
+        order: SortOrder,
+    ): List<Long> {
+        val adjustedOrder = if (order is SortOrder.UseCollectionOrdering) {
+            @Suppress("DEPRECATION")
+            SortOrder.BuiltinSortKind(
+                get_config("noteSortType", null as String?) ?: "noteFld",
+                get_config("sortBackwards", false) ?: false,
```
Comment (dae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/CollectionV16.kt:167]: Sorry, I should have noticed this last time:
```suggestion
                get_config("browserNoteSortBackwards", false) ?: false,
```

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.kt [hunk 4]
```diff
@@ -1349,8 +1349,11 @@ open class Collection(
         return Finder(this).findOneCardByNote(query)
     }
 
-    /** Return a list of note ids  */
-    fun findNotes(query: String?): List<Long> {
+    /** Return a list of note ids
+     * @param order only used in overridden V16 findNotes() method
+     * */
+    @JvmOverloads
+    open fun findNotes(query: String?, order: SortOrder = SortOrder.NoOrdering()): List<Long> {
```
Comment (dae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Collection.kt:1356]: Why is query nullable?

Comment (viciousAegis) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Collection.kt:1356]: That was my bad, I mixed up two different function parameters

**Comment by mikehardy** [comment]
Verified both of dae's comments are addressed, thanks all for the review and @viciousAegis for the work - :rocket:

**Pull request merged**